aten/src/ATen/native/TensorConversions.cpp

```diff
@@ -19,13 +19,25 @@ static inline Device ensure_has_index(Device device) {
   return impl->getDevice();
 }
 
-static inline Tensor to_impl(const Tensor& self, const TensorOptions& options, bool non_blocking) {
-  auto r = at::empty(self.sizes(), options);
+static inline Tensor to_impl(const Tensor& self, const TensorOptions& options, bool non_blocking, c10::optional<c10::MemoryFormat> optional_memory_format) {
+  auto memory_format =
+      optional_memory_format.value_or(MemoryFormat::Contiguous);
```

Comment:
This one will change later right?

Reply:
Yes, now we need to introduce additional argument and change codebase to use `.to(..., memory_format=contig)` wherever  developers counting on old `.to` side-effect.